### PR TITLE
AST Node cloning

### PR DESCRIPTION
**Pull request by robin-samuel**
Finalized the node cloning process. Working with optional nodes and the interfaces. Also handled the missing pointer error.

### CODE REVIEW

**Comment by T14Raptor** [comment]
Please resolve conflicts

**Comment by robin-samuel** [comment]
should be done

File: ast/gen_clone.go [hunk 1]
```diff
@@ -250,3 +399,66 @@ func newFieldList(name string, t ast.Expr) *ast.FieldList {
...
+							},
+						},
+					},
+				},
+			},
+		}
+		cases = append(cases, caseClause)
+	}
+
+	defaultCase := &ast.CaseClause{
```
Comment (T14Raptor) [inline ast/gen_clone.go]: Should be able to remove this redundant default case

File: ast/gen_clone.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ import (
 	"log"
 	"os"
 	"slices"
+	"strings"
 )
 
 // Generates visit.go
```
Comment (T14Raptor) [inline ast/gen_clone.go]:
```suggestion
// Generates clone.go
```

File: ast/gen_clone.go [hunk 3]
```diff
@@ -250,3 +399,66 @@ func newFieldList(name string, t ast.Expr) *ast.FieldList {
...
+
+	defaultCase := &ast.CaseClause{
+		List: nil,
+		Body: []ast.Stmt{},
+	}
+
+	switchStmt := &ast.TypeSwitchStmt{
+		Assign: &ast.AssignStmt{
+			Lhs: []ast.Expr{
+				ast.NewIdent(strings.ToLower(intf.Name)),
```
Comment (T14Raptor) [inline ast/gen_clone.go]: Could you make only the first letter lowercase? Would look better

**Review by T14Raptor — CHANGES_REQUESTED** [review]

**Review by T14Raptor — APPROVED** [review]

**Pull request merged**